chainercv/experimental/links/model/fcis/utils/mask_voting.py

```diff
@@ -0,0 +1,170 @@
+import numpy as np
+
+from chainercv.transforms.image.resize import resize
+from chainercv.utils.bbox.bbox_iou import bbox_iou
+from chainercv.utils import non_maximum_suppression
+
+
+def mask_aggregation(
+        bbox, mask_score, mask_weight,
+        size, binary_thresh
+):
+    assert bbox.shape[0] == len(mask_score)
+    assert bbox.shape[0] == mask_weight.shape[0]
+
+    mask = np.zeros(size, dtype=np.float32)
+    for bb, msk_sc, msk_w in zip(bbox, mask_score, mask_weight):
+        bb = np.round(bb).astype(np.int32)
+        y_min, x_min, y_max, x_max = bb
+        if y_max - y_min > 0 and x_max - x_min > 0:
+            msk_sc = resize(
+                msk_sc.astype(np.float32)[None],
+                (y_max - y_min, x_max - x_min))
+            msk_m = (msk_sc >= binary_thresh).astype(np.float32)[0]
+            mask[y_min:y_max, x_min:x_max] += msk_m * msk_w
+
+    y_indices, x_indices = np.where(mask >= binary_thresh)
+    if len(y_indices) == 0 or len(x_indices) == 0:
+        return None, None
+    else:
+        y_max = y_indices.max() + 1
+        y_min = y_indices.min()
+        x_max = x_indices.max() + 1
+        x_min = x_indices.min()
+
+        c_bbox = np.array(
+            [y_min, x_min, y_max, x_max],
+            dtype=np.float32)
+        c_mask = mask[y_min:y_max, x_min:x_max]
+        return c_mask[None], c_bbox[None]
+
+
+def mask_voting(
+        mask_score, bbox, score, size,
```

Comment:
I have two comments on the names of variables.

First, if the range of the value is `[0, 1]`, use `_prob` instead of `_score`.

Second, the name `mask_scores` can be misunderstood as an array with shape `(R, H, W)`. How about naming a variable `(R, RH, RW)` as  `roi_mask_prob`.  (depending on the range of its values). This name is consistent with names in `fcis.py`.
Also, how about renaming `bbox` to `roi` so that it is clear that `roi_mask_prob` is defined in the region of `roi`.
When iterating over `roi`, either `for bb in roi:` or `for r in roi` is OK.
In addition to that, `score` can be renamed to `roi_prob` to make its shape more precise (this modification is only optional).

Reply:
You actually do use something similar to what I suggested in `test_mask_voting.py`.

Reply:
> I look in some chainercv codes, and score and prob are mixed.
For example, object_detection returns scores, which is probability.
In order to clean up, we first set the rule first.

I missed this comment. sorry.

Reply:
I totally agree that `prob` is easier to understand.
But, for API, `score` is consistent to object detection API, so I will use `scores` :(.

Reply:
You can consistently use `prob` up until the end of `prediction`.
In the end, you do `scores.append(prob)`.

Reply:
OK